pkg/apis/healthchecking/v1alpha1/machinehealthcheck_types.go

```diff
@@ -36,6 +36,9 @@ type MachineHealthCheckList struct {
 
 // MachineHealthCheckSpec defines the desired state of MachineHealthCheck
 type MachineHealthCheckSpec struct {
+	// Remediation strategy to use in case of problem detection
+	// default is machine deletion
+	Remediation string `json:"remediation"`
```

Comment:
Nit: Can we call this `remediatonStrategy`? It fits with things like `updateStrategy` in `Deployment` objects.

Reply:
Also, maybe this should be a `type RemediationStrategyType string`? That will let us pass strategy type around with some guarantees beyond it being a string.